### PR TITLE
Refactor veff

### PR DESCRIPTION
**Pull request by cg-laser**
the veff and aeff for surface muons functions are combined as they did 99% the same stuff.

### CODE REVIEW

File: NuRadioMC/utilities/Veff.py [hunk 1]
```diff
@@ -468,38 +222,48 @@ def get_Veff_single(filename, trigger_names, trigger_names_dict, trigger_combina
         fin.attrs['phimin']
```
Comment (christophwelling) [inline NuRadioMC/utilities/Veff.py:222]: It looks like this should be phimin = fin.attrs['phimin']
Same in line 224

File: NuRadioMC/utilities/Veff.py [hunk 2]
```diff
@@ -468,38 +222,48 @@ def get_Veff_single(filename, trigger_names, trigger_names_dict, trigger_combina
...
         fin.attrs['phimax']
-    V = fin.attrs['volume']
+    if(veff_aeff == "veff"):
+        volume_proj_area = fin.attrs['volume']
+    elif(veff_aeff == "aeff_surface_muons"):
+        area = fin.attrs['area']
+        # The used area must be the projected area, perpendicular to the incoming
+        # flux, which leaves us with the following correction. Remember that the
+        # zenith bins must be small for the effective area to be correct.
+        volume_proj_area = area * 0.5 * (np.abs(np.cos(thetamin)) + np.abs(np.cos(thetamax)))
```
Comment (christophwelling) [inline NuRadioMC/utilities/Veff.py:232]: Shouldn't this be np.abs(...) - np.abs(...)  ?

File: NuRadioMC/utilities/Veff.py [hunk 3]
```diff
@@ -557,7 +321,13 @@ def get_Veff_single(filename, trigger_names, trigger_names_dict, trigger_combina
                     triggered = triggered & (np.array(fin[f'station_{station:d}/ray_tracing_reflection'])[..., max_amps, 0][:, 0] == values['n_reflections'])
 
             triggered = remove_duplicate_triggers(triggered, fin['event_group_ids'])
-            Veff = V * np.sum(weights[triggered]) / n_events
+            Veff = volume_proj_area * np.sum(weights[triggered]) / n_events
```
Comment (christophwelling) [inline NuRadioMC/utilities/Veff.py:324]: I guess this depends on how exactly effective volume is defined, but do we have to divide by number of events or by the sum of all weights? I.e. is absorption in the earth considered part of the effective volume? Just asking to be sure

File: NuRadioMC/utilities/Veff.py [hunk 4]
```diff
@@ -610,29 +380,29 @@ def get_Veff_single(filename, trigger_names, trigger_names_dict, trigger_combina
...
-            Veff_low = V * FC_low / n_events
-            Veff_high = V * FC_high / n_events
-            out['Veffs'][trigger_name] = [Veff, Vefferror, np.sum(weights[triggered]), Veff_low, Veff_high]
+            out[veff_aeff][trigger_name] = [Veff, Vefferror, np.sum(weights[triggered]), Veff_low, Veff_high]
     return out
 
 
 def tmp(args):
-    return get_Veff_single(*args)
+    return get_Veff_Aeff_single(*args)
```
Comment (christophwelling) [inline NuRadioMC/utilities/Veff.py:396]: Why is this function necessary? Can't you just call get_Veff_Aeff_single directly?

**Review by christophwelling — CHANGES_REQUESTED** [review]
I think there are some minor errors, other than that it looks good.

File: NuRadioMC/utilities/Veff.py (hunk 2 above)
Comment (cg-laser) [reply NuRadioMC/utilities/Veff.py:232]: no, this calculates the average zenith angle. So it is 1/2*(A+B)

Comment (christophwelling) [reply NuRadioMC/utilities/Veff.py:232]: Okay, I thought it was supposed to be an integral

File: NuRadioMC/utilities/Veff.py (hunk 3 above)
Comment (cg-laser) [reply NuRadioMC/utilities/Veff.py:324]: you always divide by the number of events. If Earth attenuation is considered the nominator is the sum of the triggered events times their weight.

File: NuRadioMC/utilities/Veff.py (hunk 1 above)
Comment (cg-laser) [reply NuRadioMC/utilities/Veff.py:222]: yes, thanks

File: NuRadioMC/utilities/Veff.py (hunk 4 above)
Comment (cg-laser) [reply NuRadioMC/utilities/Veff.py:396]: there is a tiny difference. tmp unpacks the arguments with the `*` operator

**Review by christophwelling — APPROVED** [review]

**Pull request merged**